### PR TITLE
Support pass kwargs to sd3 custom attention processor

### LINKED ISSUE #8855: "joint_attention_kwargs" don't pass the parameters to AttentionProcessor
I want to train my ipadapter in sd3. I noticed that joint_attention_kwargs is used in passing the parameters to AttentionProcessor (https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/transformers/transformer_sd3.py#L309). But it seems joint_attention_kwargs is not passed to transformer_blocks. Is my usage method wrong or this parameter has not been developed yet? Thank you!

### PR DESCRIPTION
**Pull request by Matrix53**
# What does this PR do?

Fixes #8855 as #9517 

As mentioned in #8855 and #9516, pass kwargs to custom attention processor is preferable when implementing layout control, attention replacing or introducing side effect to attention computation. Now this feature is supported by sd, sdxl, FLUX series, but not sd3. This PR fix the problem, and may support future training-free experiments in sd3 series models.

example code:
```python
class MyAttentionProcessor(JointAttnProcessor2_0):
    def __init__(self, **custom_args):
        super().__init__()
        # some initialization
    
    def __call__(self, ..., added_argu=None) -> torch.FloatTensor:
        # custom attention forward pass

my_processor = MyAttentionProcessor(**custom_args)
pipe.transformer.set_attn_processor(my_processor) # sd3 pipeline
joint_attention_kwargs = {"added_argu": 100} # prepare passing added_argu to attention processor
pipe.transformer(..., joint_attention_kwargs=joint_attention_kwargs) # custom transformer forward pass
```

The documentation already describes usage of the joint_attention_kwargs:
image

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Hello @yiyixuxu @sayakpaul, I just want to do the same to sd3 as to FLUX in #9517

### CODE REVIEW

**Review by a-r-r-o-w — COMMENTED** [review]
The PR looks good for the specific changes required to support kwargs, but all the other changes are unrelated. Please revert any changes not related supporting kwargs such as multi-line import, multi-line assigments, etc. These were not required before, nor needed now due to 120 token column limit. If these were caused by your text editor automatically, please configure it to use the same settings as pyproject.yaml for this specific repo

**Comment by Matrix53** [comment]
Thanks @a-r-r-o-w 😊, I've configured my editor and reverted all unrelated changes. Now this PR only contains necessary changes.

File: src/diffusers/models/attention.py [hunk 1]
```diff
@@ -188,7 +188,11 @@ def set_chunk_feed_forward(self, chunk_size: Optional[int], dim: int = 0):
         self._chunk_dim = dim
 
     def forward(
-        self, hidden_states: torch.FloatTensor, encoder_hidden_states: torch.FloatTensor, temb: torch.FloatTensor
+        self,
+        hidden_states: torch.FloatTensor,
+        encoder_hidden_states: torch.FloatTensor,
+        temb: torch.FloatTensor,
+        joint_attention_kwargs: Dict[str, Any] = None,
```
Comment (hlky) [inline src/diffusers/models/attention.py:195]:
```suggestion
        joint_attention_kwargs: Dict[str, Any] = {},
```

Or we can check `joint_attention_kwargs` is not None before unpacking

Comment (Matrix53) [reply src/diffusers/models/attention.py]: Thanks for review😊. I have committed this fix.

**Review by hlky — CHANGES_REQUESTED** [review]
Edit: let's add `joint_attention_kwargs = joint_attention_kwargs or {}` instead like https://github.com/huggingface/diffusers/pull/9517/files

**Review by hlky — APPROVED** [review]
Thanks @Matrix53!

**Comment by hlky** [comment]
cc @sayakpaul @yiyixuxu WDYT about making default `joint_attention_kwargs` as `{}`? We end up doing `joint_attention_kwargs or {}` anyway, there are several checks for None which are sort of unnecessary.

**Comment by sayakpaul** [comment]
Not opposed to the idea.

File: src/diffusers/models/transformers/transformer_sd3.py [hunk 2]
```diff
@@ -369,6 +369,7 @@ def forward(
             If `return_dict` is True, an [`~models.transformer_2d.Transformer2DModelOutput`] is returned, otherwise a
             `tuple` where the first element is the sample tensor.
         """
+        joint_attention_kwargs = joint_attention_kwargs or {}
```
Comment (yiyixuxu) [inline src/diffusers/models/transformers/transformer_sd3.py:372]:
```suggestion
```

**Review by yiyixuxu — COMMENTED** [review]
thanks!

**Pull request merged**